platform/bare_metal/mbed_lib.json

```diff
@@ -1,4 +1,4 @@
 {
   "name": "bare-metal",
-  "requires": ["platform", "drivers", "rtos-api", "mbed-trace"]
+  "requires": ["platform", "drivers", "rtos-api", "mbed-trace", "blockdevice"]
```

Comment:
Worth mentioning "migration action": For a bare metal application that uses block device, `"blockdevice"` (in addition to the actual type of block device) needs to be added to the `"requires"` attribute of `mbed_app.json`.

@AGlass0fMilk FYI, this will be required for [mbed-mcuboot-demo](https://github.com/AGlass0fMilk/mbed-mcuboot-demo) once it's merged and if you switch to a newest Mbed OS.